client/src/pages/Balance.jsx

```diff
@@ -0,0 +1,205 @@
+                <Grid item>
+                  <Typography className={classes.h1}>Your balance:</Typography>
+                </Grid>
+                <Grid item>
+                  <Typography className={classes.balanceText}>{user && user.balance} credits</Typography>
+                </Grid>
+              </Grid>
+            </Grid>
+            <Grid item>
+              <Grid container direction="column" alignItems="center" spacing={2}>
+                <Grid item>
+                  <Typography className={classes.h2}>Top Up:</Typography>
+                </Grid>
+                <Grid item>
+                  <Grid className={classes.input} container direction="row" justify="center" alignItems="center">
+                    <Grid item>
+                      <IndeterminateCheckBoxIcon
+                        className={refillAmount !== 1 ? classes.subtract : classes.disabled}
+                        onClick={decrement}
+                        disabled={refillAmount === 1}
+                      />
+                    </Grid>
+                    <Grid item>
+                      <Typography className={classes.counter}>{refillAmount}</Typography>
+                    </Grid>
+                    <Grid item>
+                      <AddBoxIcon
+                        className={classes.add}
+                        onClick={increment}
+                      />
+                    </Grid>
+                  </Grid>
+                </Grid>
+              </Grid>
+            </Grid>
+            <Grid item>
+              <Button className={classes.button} onClick={handleCheckout}>
+                Checkout
+              </Button>
+            </Grid>
+          </Grid>
+        ) : (
+          <Grid container direction="column" alignItems="center" spacing={2}>
+            <Grid item>
+              <Typography className={classes.h1}>Checkout</Typography>
+            </Grid>
+            <Grid item>
+              <Grid container direction="column">
+                <Grid item>
+                  <Typography className={classes.h2}>Card number</Typography>
+                </Grid>
+                <Grid item>
+                  <TextField />
+                </Grid>
+              </Grid>
+            </Grid>
+            <Grid item>
+              <Grid container direction="row" justify="center">
+                <Grid item>
+                  <Grid container direction="column">
+                    <Grid item>
+                      <Typography className={classes.h2}>Expiry date</Typography>
+                    </Grid>
+                    <Grid item>
+                      <TextField />
+                    </Grid>
+                  </Grid>
+                </Grid>
+                <Grid item>
+                  <Grid container direction="column">
+                    <Grid item>
+                      <Typography className={classes.h2}>CVC</Typography>
```

Comment:
When you change this to Stripe, you'll probably be using https://github.com/stripe/react-stripe-elements